src/features/defi/providers/fox-farming/components/FoxFarmingManager/Deposit/components/Approve.tsx

```diff
@@ -39,7 +40,9 @@ export const Approve: React.FC<FoxFarmingApproveProps> = ({ accountId, onNext })
   const { query } = useBrowserRouter<DefiQueryParams, DefiParams>()
   const { chainId, contractAddress } = query
   const opportunity = state?.opportunity
-  const { allowance, approve, getStakeGasData } = useFoxFarming(contractAddress)
+  const { allowance, approve, getStakeGasData } = useFoxFarming(
+    contractAddress as FoxEthStakingContractAddress,
```

Comment:
We could write a type guard assertion that checks if the `contractAddress` string is a `FoxEthStakingContractAddress` and avoid the type assertion.

Since we are casting here anyway, and so are confident it will be of the expected type, a type guard assertion seems reasonable.

Reply:
8acd58573b14b5ddc71ed0e95fc173e2aaec76d6